### PR TITLE
Begin producing 2.1 nightly images

### LINKED ISSUE #279: Begin producing 2.1 nightly images
Depends on microsoft/dotnet-nightly to begin consuming 2.1 daily builds of .NET Core.

cc @MichaelSimons

### PR DESCRIPTION
**Pull request by natemcmaster**
Depends on https://github.com/dotnet/dotnet-docker-nightly/pull/400

Note: CI is expected to fail until these images are up on Docker hub.

Resolves #279

### CODE REVIEW

File: README.aspnetcore-build.md [hunk 1]
```diff
@@ -12,17 +12,21 @@ This repository contains images that are used to compile/publish ASP.NET Core ap
...
     - [`1.1.2-nanoserver` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/dev/1.1/nanoserver/sdk/Dockerfile)
-- `2.0.0`, `2.0`, `2`, `latest`
+- `2.0.0`, `2.0`
     - [`2.0.0-stretch` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/dev/2.0/stretch/sdk/Dockerfile)
     - [`2.0.0-nanoserver` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/dev/2.0/nanoserver/sdk/Dockerfile)
 - [`2.0.0-jessie`, `2.0-jessie`, `2-jessie` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/dev/2.0/jessie/sdk/Dockerfile)
+- `2.1.0-preview1`, `2.1`, `2`, `latest`
+    - [`2.1.0-preview1-stretch` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/dev/2.1/stretch/sdk/Dockerfile)
+    - [`2.1.0-preview1-nanoserver` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/dev/2.1/nanoserver/sdk/Dockerfile)
+- [`2.1.0-preview1-jessie`, `2.1-jessie`, `2-jessie` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/dev/2.1/jessie/sdk/Dockerfile)
```
Comment (MichaelSimons) [inline README.aspnetcore-build.md]: Looks like the `2-jessie` tag is used twice?
` - [`2.0.0-jessie`, `2.0-jessie`, `2-jessie` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/dev/2.0/jessie/sdk/Dockerfile)`
`- [`2.1.0-preview1-jessie`, `2.1-jessie`, `2-jessie` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/dev/2.1/jessie/sdk/Dockerfile)`

File: README.aspnetcore-build.md [hunk 2]
```diff
@@ -12,17 +12,21 @@ This repository contains images that are used to compile/publish ASP.NET Core ap
 - `1.1.2`, `1.1`, `1`
     - [`1.1.2-jessie` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/dev/1.1/jessie/sdk/Dockerfile)
     - [`1.1.2-nanoserver` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/dev/1.1/nanoserver/sdk/Dockerfile)
-- `2.0.0`, `2.0`, `2`, `latest`
+- `2.0.0`, `2.0`
     - [`2.0.0-stretch` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/dev/2.0/stretch/sdk/Dockerfile)
     - [`2.0.0-nanoserver` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/dev/2.0/nanoserver/sdk/Dockerfile)
 - [`2.0.0-jessie`, `2.0-jessie`, `2-jessie` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/dev/2.0/jessie/sdk/Dockerfile)
+- `2.1.0-preview1`, `2.1`, `2`, `latest`
```
Comment (MichaelSimons) [inline README.aspnetcore-build.md]: In microsoft/dotnet-nightly the `2` and `latest` tags were left on 2.0 as that will be what the tags will point to when 2.1.0-preview1 is released as a preview.

File: README.aspnetcore.md [hunk 3]
```diff
@@ -15,10 +15,14 @@ These images contain the runtime only. Use [`microsoft/aspnetcore-build-nightly`
 - `1.1.2`, `1.1`, `1`
     - [`1.1.2-jessie` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/dev/1.1/jessie/runtime/Dockerfile)
     - [`1.1.2-nanoserver` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/dev/1.1/nanoserver/runtime/Dockerfile)
-- `2.0.0`, `2.0`, `2`, `latest`
+- `2.0.0`, `2.0`
```
Comment (MichaelSimons) [inline README.aspnetcore.md]: Same comments from build apply here.

Comment (natemcmaster) [inline README.aspnetcore.md]: At one point you mentioned that you wanted to generate this section from the manifest.json file. Did you end up building a tool to do that?

File: README.aspnetcore-build.md (hunk 2 above)
Comment (natemcmaster) [inline README.aspnetcore-build.md]: updated

File: README.aspnetcore.md (hunk 3 above)
Comment (MichaelSimons) [inline README.aspnetcore.md]: yes - one thing of note is this [announcement ](https://github.com/dotnet/dotnet-docker/issues/275) which changed the readme formet used in microsoft/dotnet

Sample usage where docker github repo path is f:\repos\dotnet-docker-nightly
`docker run --rm -v /var/run/docker.sock:/var/run/docker.sock -v f:\repos\dotnet-docker-nightly:/repo -w /repo microsoft/dotnet-buildtools-prereqs:image-builder-jessie-20170817154316 generateTagsReadme https://github.com/dotnet/dotnet-docker-nightly/blob/master`

Comment (MichaelSimons) [inline README.aspnetcore.md]: It doesn't update the actual readme file - it just writes to the console.  Could be made better but this was what I considered the min bar 😄.  Feel free to make a contribution to the tool if you have ideas.

Comment (natemcmaster) [inline README.aspnetcore.md]: It's a good start. I'm going to add a script to the repo to help me remember all those variables required to generate.

Also, it appears the manifest file in my repo needs updating....

```
READING MANIFEST
----------------
Newtonsoft.Json.JsonSerializationException: Cannot deserialize the current JSON array (e.g. [1,2,3]) into type 'System.Collections.Generic.IDictionary`2[System.String,Microsoft.DotNet.ImageBuilder.Model.Tag]' because the type requires a JSON object (e.g. {"name":"value"}) to deserialize correctly.
To fix this error either change the JSON to a JSON object (e.g. {"name":"value"}) or change the deserialized type to an array or a type that implements a collection interface (e.g. ICollection, IList) like List<T> that can be deserialized from a JSON array. JsonArrayAttribute can also be added to the type to force it to deserialize from a JSON array.
Path 'repos[0].images[0].sharedTags', line 20, position 25.
```

Comment (MichaelSimons) [inline README.aspnetcore.md]: Ah yes - sorry about that - I needed to make a schema change in order to support ARM.  [Here ](https://github.com/dotnet/dotnet-docker-nightly/commit/a1dcba51294abadf17a07c353e3586c464984143#diff-4b1eb3dc48c4e16d49db5b42298fe654L24) is a sample of how you would need to change the schema.

Comment (natemcmaster) [inline README.aspnetcore.md]: Ok, I'm going to do that in a separate PR to 'master'. Looks like a significant change.
